crates/fluvio-spu/src/replication/test.rs

```diff
@@ -861,3 +871,82 @@ async fn test_sync_2_replicas_but_one_reject() {
 
     spu_server.notify();
 }
+
+#[fluvio_future::test(ignore)]
+async fn test_sync_larger_records() {
+    let num_records = 5;
+    let port = portpicker::pick_unused_port().expect("No free ports left");
+    let builder = TestConfig::builder()
+        .followers(1_u16)
+        .base_port(port)
+        .generate("sync_larger_records");
+
+    let leader_gctx = builder.leader_ctx().await;
+
+    let spu_server = create_internal_server(builder.leader_addr(), leader_gctx.clone()).run();
+
+    let replica = builder.replica();
+
+    let actions = leader_gctx
+        .apply_replica_update(UpdateReplicaRequest::with_all(1, vec![replica.clone()]))
+        .await;
+    assert!(actions.is_empty());
+
+    // give leader controller time to startup
+    sleep(Duration::from_millis(MAX_WAIT_LEADER)).await;
+
+    let leader = leader_gctx
+        .leaders_state()
+        .get(&replica.id)
+        .await
+        .expect("replica");
+    assert!(leader_gctx
+        .followers_state()
+        .get(&replica.id)
+        .await
+        .is_none());
+    // should be new
+    assert_eq!(leader.leo(), 0);
+    assert_eq!(leader.hw(), 0);
+
+    // create a raw record with 512kb
+    let record_test: Vec<u8> = vec![10; 512 * 1024];
+
+    // write a batch with 5 records with 512kb, total of 2.5MB
+    leader
+        .write_record_set(
+            &mut create_raw_recordset_inner(num_records, &record_test),
+            leader_gctx.follower_notifier(),
+        )
+        .await
+        .expect("write");
+
+    assert_eq!(leader.leo(), num_records as i64);
+    assert_eq!(leader.hw(), 0);
+
+    let follower_gctx = builder.follower_ctx(0).await;
+    let actions = follower_gctx
+        .apply_replica_update(UpdateReplicaRequest::with_all(1, vec![replica.clone()]))
+        .await;
+    assert!(actions.is_empty());
+    let follower = follower_gctx
+        .followers_state()
+        .get(&replica.id)
+        .await
+        .expect("follower");
+    assert_eq!(follower.leader(), LEADER);
+    assert_eq!(follower.leo(), 0);
+    assert_eq!(follower.hw(), 0);
+
+    // wait until follower sync up with leader
+    sleep(Duration::from_millis(*MAX_WAIT_REPLICATION)).await;
+    assert_eq!(follower.leo(), num_records as i64);
+
+    // hw has been replicated
+    assert_eq!(follower.hw(), num_records as i64);
+    assert_eq!(leader.hw(), num_records as i64);
```

Comment:
check records from leader and follower is same

Reply:
done